evalml/data_checks/target_leakage_data_check.py

```diff
@@ -63,14 +62,17 @@ def validate(self, X, y):
         y = _convert_to_woodwork_structure(y)
         X = _convert_woodwork_types_wrapper(X.to_dataframe())
         y = _convert_woodwork_types_wrapper(y.to_series())
-
-        if y.dtype not in numeric_and_boolean_dtypes:
+        combined = X.copy()
```

Comment:
Could we remove the overhead of converting X and y above and just convert using a combined dataframe?

Reply:
@jeremyliweishih I tried simplifying this, but I think this might be the best way to preserve the functionality we want. In order to properly combine, we either have to have ww.Datatables or pd.Dataframes, so the first 2 '_convert_to_woodwork_structure` calls make's that process cleaner. The next two calls preserve the types that woodwork gives, while allowing us to then copy X and combine it with y, and the last call allows us to turn it back to ww. If I combined first, I would need to convert both the X and y to pandas Dataframes/Series, which could lead to longer code. Definitely free to discuss this more if I'm missing something!